core/src/test/java/org/apache/iceberg/hadoop/HadoopFileIOTest.java

```diff
@@ -203,6 +206,43 @@ public void testJsonParserWithHadoopConf() throws Exception {
     testJsonParser(hadoopFileIO, tempDir);
   }
 
+  @Test
+  public void fileIOWithStorageCredentialsKryoSerialization() throws IOException {
+    this.hadoopFileIO = new HadoopFileIO();
+    hadoopFileIO.setCredentials(
+        ImmutableList.of(
+            StorageCredential.create("prefix", Map.of("key1", "val1", "key2", "val2"))));
+    hadoopFileIO.initialize(Map.of());
+
+    assertThat(TestHelpers.KryoHelpers.roundTripSerialize(hadoopFileIO).credentials())
+        .isEqualTo(hadoopFileIO.credentials());
+  }
+
+  @Test
+  public void fileIOWithStorageCredentialsJavaSerialization()
+      throws IOException, ClassNotFoundException {
+    this.hadoopFileIO = new HadoopFileIO();
+    hadoopFileIO.setCredentials(
+        ImmutableList.of(
+            StorageCredential.create("prefix", Map.of("key1", "val1", "key2", "val2"))));
+    hadoopFileIO.initialize(Map.of());
+
+    assertThat(TestHelpers.roundTripSerialize(hadoopFileIO).credentials())
+        .isEqualTo(hadoopFileIO.credentials());
+  }
+
+  @Test
+  public void testPassCredentials() {
```

Comment:
this should rather be testing that the settings in the hadoop conf are overwritten with whatever gets passed via storage credentials

Reply:
> this should rather be testing that the settings in the hadoop conf are overwritten with whatever gets passed via storage credentials

Great suggestion, thanks! The test has been updated accordingly.